src/main/java/io/mathdojo/QuestionFunction.java

```diff
@@ -11,25 +15,128 @@
 
 @SpringBootApplication
 public class QuestionFunction {
+	/**
+	 * This class contains all the functions that can be called in this service.
+	 * Functions are called through RequestHandler classes
+	 */
 
 	@Autowired
 	public MathDojoQuestionRepository repository;
-	private final Question emptyDatabaseReturnValue = new Question("the database is empty", null, null, null, 0, null, false);
+	@Autowired
+	public MathDojoTopicRepository tRepository;
 
 	public static void main(String[] args) throws Exception {
 		SpringApplication.run(QuestionFunction.class, args);
 	}
 
 	@Bean
-	public Function<GetObject, Question> getQuestion() {
-		//customise the query here
-		return get ->  Iterables.getLast(repository.findAll() , emptyDatabaseReturnValue) ;
+	public Function<Question, Question> getQuestion() {
+		return input -> Iterables.getFirst(
+				repository.findByQuestionTitle(input.getQuestionTitle()).stream()
+						.filter(i -> i.getDifficulty().equals(input.getDifficulty())).collect(Collectors.toList()),
+				Question.EMPTY_DATABASE);
+	}
+
+	@Bean
+	public Consumer<Question> createQuestion() {
+		return question -> repository.save(question);
+	}
+
+	@Bean
+	public Function<Question, Question> getQuestionById() {
+		return question -> repository.findById(question.getId()).isPresent()
+				? repository.findById(question.getId()).get()
+				: Question.EMPTY_DATABASE;
+	}
+
+	@Bean
+	public Consumer<Question> updateQuestion() {
+		return new Consumer<Question>() {
+
+			@Override
+			public void accept(Question question) {
+				Question oldQuestion = repository.findById(question.getId()).isPresent()
+						? repository.findById(question.getId()).get()
+						: null;
```

Comment:
I think if someone is trying to update a question that doesn't exist we should throw an error. The problem with this implementation is that could allow someone to create a new question via the `PUT` method which is not our intention.